python/ray/util/iter.py

```diff
@@ -680,6 +719,44 @@ def add_wait_hooks(it):
             self.local_transforms + [apply_foreach],
             name=self.name + ".for_each()")
 
+    def for_each_concur(self,
+                        fn: Callable[[T], U],
+                        max_concur=2,
+                        resources=None) -> "LocalIterator[U]":
+        if resources is None:
+            resources = {}
+
+        def apply_foreach_concur(it):
+            cur = []
+            remote = ray.remote(fn).options(**resources)
+            remote_fn = remote.remote
+            for item in it:
+                if isinstance(item, _NextValueNotReady):
+                    yield item
+                else:
+                    finished, remaining = ray.wait(cur, timeout=0)
+                    if len(remaining) >= max_concur:
+                        ray.wait(cur, num_returns=(len(finished) + 1))
+                    cur.append(remote_fn(item))
+
+                    while True:
+                        to_yield = cur[0]
+                        finished, remaining = ray.wait([to_yield], timeout=0)
+                        if finished:
+                            cur.pop(0)
+                            yield ray.get(to_yield)
+                        else:
+                            break
+
+            yield from ray.get(cur)
+
+        # TODO: What does add_wait_hooks do in the regular for_each?
```

Comment:
please leave your github account in the TODO so others know who to ping if they have questions